packages/http/fetch/src/middlewares/options/userAgentHandlerOptions.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { RequestOption } from "@microsoft/kiota-abstractions";
+
+export const UserAgentHandlerOptionsKey = "UserAgentHandlerOptionKey";
+
+export class UserAgentHandlerOptions implements RequestOption {
+	getKey(): string {
+		return UserAgentHandlerOptionsKey;
+	}
+	/**
+	 * @public
+	 * @constructor
+	 * To create an instance of UserAgentHandlerOption
+	 * @param {boolean} [enable = true] - Whether to add the user agent header to the request
+	 * @param {string} [productName = "kiota-typescript"] - The product name to be added to the user agent header
+	 * @param {string} [productVersion = "1.0.0-preview.12"] - The product version to be added to the user agent header
+	 */
+	public constructor(public enable: boolean = true, public productName: string = "kiota-typescript", public productVersion: string = "1.0.0-preview.12") {}
```

Comment:
This feels prone to falling out of date.
Can write the version from package.json into a constant on build and use that value?
We have a process that does this during build for mgt

Reply:
sure, do you have pointers to share?

Reply:
nevermind, just pushed a script, thanks for calling this out.

Reply:
ah, I hate that GitHub doesn't refresh the comments live. Didn't want to put my hands in gulp, I figured a powershell script would do the trick.